zng/builder_test.go

```diff
@@ -0,0 +1,68 @@
+package zng_test
+
+import (
+	"bytes"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zio/zngio"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/stretchr/testify/assert"
+)
+
+const builder = `
+#0:record[key:ip]
+#1:record[a:int64,b:int64,c:int64]
+#2:record[a:int64,r:record[x:int64]]
+0:[1.2.3.4;]
+1:[1;2;3;]
+2:[7;[3;]]`
+
+func TestBuilder(t *testing.T) {
+	in := []byte(strings.TrimSpace(builder) + "\n")
+	r := zngio.NewReader(bytes.NewReader(in), resolver.NewContext())
```

Comment:
zngio.Reader is nicely flexible about leading and trailing newlines, so you can simplify if you like.
```suggestion
	r := zngio.NewReader(strings.NewReader(builder), resolver.NewContext())
```